src/wallet.js

```diff
@@ -542,10 +596,43 @@ export class Wallet extends React.Component {
     });
   }
 
+  signAndSendTransaction(closeOnSuccess) {
+    this.runModal('Sending Transaction', 'Please wait...', async () => {
+      let signature = '';
+      try {
+        signature = await web3.sendAndConfirmTransaction(
+          this.web3sol,
+          this.state.unsignedTransaction,
+          this.state.account,
+        );
+      } catch (err) {
+        // Transaction failed but fees were still taken
+        this.setState({
+          balance: await this.web3sol.getBalance(this.state.account.publicKey),
+        });
+        this.postWindowMessage('sendCustomTransactionResponse', {err: true});
+        throw err;
+      }
+
+      this.postWindowMessage('sendCustomTransactionResponse', {signature});
+      if (closeOnSuccess) {
+        window.close();
+      } else {
+        this.setState({
+          balance: await this.web3sol.getBalance(this.state.account.publicKey),
+        });
+      }
+    });
+  }
+
   confirmTransaction() {
```

Comment:
Please revert your changes here in this method

Reply:
done